### PR TITLE
Fix comparison of invalid values.

### PR DESCRIPTION
**Pull request by jheer**
Continuation of #74.

### CODE REVIEW

**Comment by jheer** [comment]
@arvind and @kanit, can you review and confirm that this PR meets your needs? It will separate invalid (`null`, `undefined` and `NaN`) values from valid values, but is not concerned with the sort order among these invalid values.

**Comment by kanitw** [comment]
This fixes examples we have in github.com/vega/vega-lite/issues/1412

(Old):
vega_editor

(Building Vega with this branch): 
vega_editor


Meanwhile, I wonder if not having order among `null` and `NaN` would cause problem in a rare case where we have both `NaN` and `null`.  So I generate this toy data that has a field with null, numbers and strings and parse it as `number`. Looks like something is still wrong:

spec: http://pastebin.com/Q5FkmUXe

vega_editor


Even if I `parse` the data as `string`, it's still incorrect. 

vega_editor

(For the latter case, I wonder if the problem is due to `stack` transform's `sortby`)

**Comment by jheer** [comment]
Thanks @kanitw! After some thought, I think we can treat `null` and `undefined` as equivalent, but need to treat `NaN` as distinct. While mixing of `null` and `NaN` values is not the ideal situation, `NaN` can arise in the midst of a transform pipeline due to faulty calculations, etc, and so should be handled separately from otherwise "missing" values.

This PR now contains an updated comparison method; the ascending sort order is: `null`/`undefined`, `NaN`, then "valid" values.

**Comment by kanitw** [comment]
Thank you.  I just verified.   This spec now works correctly 

spec: http://pastebin.com/Q5FkmUXe

vega_editor

Note that the later case is still incorrect – but I think it might be a bug at Vega-level, so I'm filing this bug there https://github.com/vega/vega/issues/590.

spec: http://pastebin.com/ThRcjRFG
data: https://gist.githubusercontent.com/kanitw/59986cffa2f9bf36871ca07ab3e28c3e/raw/50b22fa5dfd3b5794de62856a2b958c1103b8dc6/data2.json

(This spec parse the same field that mixes null, numbers and strings as `string`)

vega_editor

If you agree that the bug in the latter case is a different problem I think we can merge and make a release :)

File: src/util.js [hunk 1]
```diff
@@ -190,30 +190,30 @@ u.comparator = function(sort) {
...
-  } else if (b === null) {
-    return 1;
-  }
-  return NaN;
+  return (a < b || a == null) && b != null ? -1 :
+    (a > b || b == null) && a != null ? 1 :
+    ((b = b instanceof Date ? +b : b),
+     (a = a instanceof Date ? +a : a)) !== a && b === b ? -1 :
+    b !== b && a === a ? 1 : 0;
+  // return a < b && b !== null ? -1 :                              // is a < b?
```
Comment (kanitw) [inline src/util.js]: looks like you amend and force push again .. not sure if you intend to keep these commented code?

Comment (jheer) [inline src/util.js]: No, that was an error. I've updated with a new force push and will merge soon...

**Pull request merged**